app/scripts/controllers/incoming-transactions.js

```diff
@@ -267,21 +269,33 @@ export default class IncomingTransactionsController {
       etherscanTransaction.isError === '0'
         ? TRANSACTION_STATUSES.CONFIRMED
         : TRANSACTION_STATUSES.FAILED;
+    const txParams = {
+      from: etherscanTransaction.from,
+      gas: bnToHex(new BN(etherscanTransaction.gas)),
+      nonce: bnToHex(new BN(etherscanTransaction.nonce)),
+      to: etherscanTransaction.to,
+      value: bnToHex(new BN(etherscanTransaction.value)),
+    };
+
+    if (etherscanTransaction.gasPrice) {
+      txParams.gasPrice = bnToHex(new BN(etherscanTransaction.gasPrice));
+    } else if (etherscanTransaction.maxFeePerGas) {
```

Comment:
Do we at all need to consider the case where one of `maxFeePerGas` or `maxPriorityFeePerGas` are truthy and the other is not?

Reply:
Also, `etherscanTransaction.maxFeePerGas` will never be of type number and equal to 0 right?

Reply:
A transaction isn't valid if it has one or the other, per the JSON-rpc spec it would be immediately rejected. 


https://github.com/ethereum/eth1.0-specs/blob/master/json-rpc/spec.json#L1709-L1730

So like, the new fields appear to be missing a 'type' parameter, also I just noticed it *does* specify WEI not GWEI i assume that we are then converting the gasPrice (also in wei) elsewhere in our code for display to the user?

Reply:
> i assume that we are then converting the gasPrice (also in wei) elsewhere in our code for display to the user?
yes, we are

Reply:
for example, in `metamask-extension/ui/components/app/transaction-breakdown/transaction-breakdown.component.js`

Reply:
see the use of the `CurrencyDisplay` component